### ISSUE TITLE
make install-bin on a Mac can fail to install text-icu

### ISSUE THREAD

**Issue by halfaya** [agda/agda#2939]
On a Mac `brew install icu4c` installs icu into a non-standard location (/usr/local/opt/icu4c/).
As a result, `make install-bin` fails when trying to install the `text-icu` package (required due to `-fenable-cluster-counting`) which relies on standard locations for its C dependencies.
```
Missing dependencies on foreign libraries:
    * Missing C libraries: icuuc, icui18n, icudata
    This problem can usually be solved by installing the system packages that
    provide these libraries (you may need the "-dev" versions). If the libraries
    are already installed but in a non-standard location then you can use the
    flags --extra-include-dirs= and --extra-lib-dirs= to specify where they are.
```
As noted these flags need to be added to `cabal install` somehow:
```
--extra-lib-dirs=/usr/local/opt/icu4c/lib
--extra-include-dirs=/usr/local/opt/icu4c/include
```
We should determine the best way to either automatically add these for Mac users, or to document how Mac users can add them manually when running `make install-bin`.

Note that this problem has also appeared for other Haskell packages (see for example https://github.com/yi-editor/yi/issues/795).

**Comment by asr** [comment]
The issue was reported from a developer point of view (i.e. using `make install-bin`). You can pass extra arguments to `cabal install` using the `CABAL_OPTS` variable defined in the `Makefile`.

@halfaya, does the following command fix the issue

```
 $ make install-bin CABAL_OPTS='--extra-lib-dirs=/usr/local/opt/icu4c/lib --extra-include-dirs=/usr/local/opt/icu4c/include
```

?

I guess end users also have the issue when they enable the `-fenable-cluster-counting` flag (this flag is enabled by default in the upstream version but it is disabled by default in the released version), so we also need to document how to fix the issue when using 

```
$ cabal install -f enable-cluster-counting
```

**Comment by halfaya** [comment]
This looks like it should be fine, although I can't reproduce now since `text-icu` has been installed so it won't fail again--I guess I could try to manually uninstall but that I recall that's a little tricky with Haskell packages. I think it should be fine to just document this as an option for Mac users.

**Comment by UlfNorell** [comment]
Fixed by @halfaya in 9b8c0d770a.

**Comment by asr** [comment]
Back-ported into 2.5.4.3.
